### PR TITLE
JAMES-2939 Fix unstable DefaultMailboxesProvisioningFilterTest

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/DefaultMailboxesProvisioningFilter.java [hunk 1]
```diff
@@ -114,7 +115,7 @@ private void createMailbox(MailboxPath mailboxPath, MailboxSession session) {
                 subscriptionManager.subscribe(session, mailboxPath.getName());
             }
             LOGGER.info("Provisioning {}. {} created.", mailboxPath, mailboxId);
-        } catch (MailboxExistsException e) {
+        } catch (MailboxExistsException | InboxAlreadyCreated e) {
```
Comment (rouazana) [inline server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/DefaultMailboxesProvisioningFilter.java]: InboxAlreadyCreated should be a subtype of MailboxExistsException, no?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (default) on project james-server-jmap-draft: You have 1 Checkstyle violation. -> [Help 1]
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**